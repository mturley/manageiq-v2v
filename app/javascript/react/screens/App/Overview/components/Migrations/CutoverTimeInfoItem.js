import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'patternfly-react';
import { formatDateTime } from '../../../../../../components/dates/MomentDate';
import ListViewTable from '../../../common/ListViewTable/ListViewTable';

const CutoverTimeInfoItem = ({ plan }) => {
  if (
    !plan.options.config_info.warm_migration ||
    (plan.options.config_info.warm_migration && !plan.options.config_info.warm_migration_cutover_datetime)
  ) {
    return null;
  }

  return (
    <ListViewTable.InfoItem key={`${plan.id}-cutoverTime`} style={{ textAlign: 'left' }}>
      <Icon type="fa" name="clock-o" />
      {__('Cutover scheduled')}
      <br />
      {formatDateTime(plan.options.config_info.warm_migration_cutover_datetime)}
    </ListViewTable.InfoItem>
  );
};

CutoverTimeInfoItem.propTypes = {
  plan: PropTypes.object
};

export default CutoverTimeInfoItem;
