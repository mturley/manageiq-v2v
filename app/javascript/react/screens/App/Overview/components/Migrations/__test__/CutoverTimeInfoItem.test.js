import React from 'react';
import { shallow } from 'enzyme';
import CutoverTimeInfoItem from '../CutoverTimeInfoItem';

describe('When initialized with warm migration plan', () => {
  const plan = {
    options: { config_info: { warm_migration: true, warm_migration_cutover_datetime: '2020-02-07T11:52:45-06:00' } }
  };
  const baseProps = {
    plan
  };

  test('renders info item', () => {
    const wrapper = shallow(<CutoverTimeInfoItem {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('When initialized with cold migration plan', () => {
  const plan = {
    options: { config_info: {} }
  };
  const baseProps = {
    plan
  };

  test('returns null', () => {
    const wrapper = shallow(<CutoverTimeInfoItem {...baseProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
