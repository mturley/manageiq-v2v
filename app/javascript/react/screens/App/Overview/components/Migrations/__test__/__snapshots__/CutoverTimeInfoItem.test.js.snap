// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When initialized with cold migration plan returns null 1`] = `""`;

exports[`When initialized with warm migration plan renders info item 1`] = `
<ListViewTableInfoItem
  className=""
  key="undefined-cutoverTime"
  stacked={false}
  style={
    Object {
      "textAlign": "left",
    }
  }
>
  <Icon
    name="clock-o"
    type="fa"
  />
  Cutover scheduled
  <br />
  February 7th 2020, 5:52 pm
</ListViewTableInfoItem>
`;
